libraries/AC_AttitudeControl/AC_AttitudeControl.cpp

```diff
@@ -635,57 +666,47 @@ float AC_AttitudeControl::sqrt_controller(float error, float p, float second_ord
     }
 }
 
-void AC_AttitudeControl::get_rotation_vehicle_to_ned(Matrix3f& m)
-{
-    m = _ahrs.get_rotation_body_to_ned();
-}
-
-void AC_AttitudeControl::get_rotation_ned_to_vehicle(Matrix3f& m)
+// Inverse proportional controller with piecewise sqrt sections to constrain second derivative
+float AC_AttitudeControl::stopping_point(float first_ord_mag, float p, float second_ord_lim)
 {
-    get_rotation_vehicle_to_ned(m);
-    m = m.transposed();
-}
-
-void AC_AttitudeControl::get_rotation_reference_to_ned(Matrix3f& m)
-{
-    m.from_euler(_att_target_euler_rad.x,_att_target_euler_rad.y,_att_target_euler_rad.z);
-}
-
-void AC_AttitudeControl::get_rotation_ned_to_reference(Matrix3f& m)
-{
-    get_rotation_reference_to_ned(m);
-    m = m.transposed();
-}
+    if (second_ord_lim < 0.0f || is_zero(second_ord_lim) || is_zero(p)) {
+        return first_ord_mag/p;
```

Comment:
Hi Oxinarf,
I didn't reply to this one because I needed to fix it first.
I had to check for multiple combinations of conditions here but I have fixed it now. This will save a bunch of repeated code in the WPNav library. Randy and I will probably move this to a maths_control library or something.